### DISCUSSION TITLE
Is it possible to add 2 PlanningSolution annotated class in a single project?

### DISCUSSION THREAD

**Discussion by yush-sys** [TimefoldAI/timefold-solver#905] [General]
I am currently working on a project very much like meeting scheduling and I have to run the solver twice. The planning entity changes for the 2 solver, so I changed the planning solutions too and made the config.xml separate. I followed this blog to add multiple solver https://timefold.ai/blog/multiple-solver-managers but the spring boot application shows this error when I try to run it.
![Screenshot 2024-06-24 093019](https://github.com/TimefoldAI/timefold-solver/assets/173650578/fb154563-50ed-40fb-963d-17e361a67dc4)

**Comment by Christopher-Chianelli** [comment]
Yes it is possible to use multiple `PlanningSolution` classes in a single Spring boot project. However, in order to do so, you must use `@Named` solver managers with `solver-config-xml` specified in the `applications.properties`:

In your `application.properties`:

```properties
timefold.solver.teacherSolver.solver-config-xml=teachersConfig.xml
timefold.solver.roomSolver.solver-config-xml=roomsConfig.xml
```

In your Java code:

```java
    @Autowired
    @Qualifier("teacherSolver")
    SolverManager<TeacherToLessonSchedule, String> teacherToLessonScheduleSolverManager;

    @Autowired
    @Qualifier("roomSolver")
    SolverManager<Timetable, String> lessonToRoomTimeslotSolverManager;
```

**Reply by yush-sys** [reply]
I had the xml in the resource and created the solver with out the @autowired.

**Reply by Christopher-Chianelli** [reply]
You are missing the `timefold.` prefix in your `application.properties`. It should look like this instead:

```properties
timefold.solver.workOrder.solver-config-xml=solver/workOrder/KsystemScheduleManagementApplicationSolverConfig.xml
timefold.solver.workOrderReverse.solver-config-xml=solver/workOrderReverse/ReverseKsystemScheduleManagementApplicationSolverConfig.xml
```

**Reply by yush-sys** [reply]
thank you.
